modules/CircularLayout/src/main/java/org/gephi/layout/plugin/circularlayout/dualcirclelayout/DualCircleLayout.java

```diff
@@ -0,0 +1,341 @@
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.InnerNodeCount.name"),
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.Category.NodePlacement.name"),
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.InnerNodeCount.desc"),
+                           "getInnerNodeCount", "setInnerNodeCount"));
+         properties.add(LayoutProperty.createProperty(
+                           this, String.class,
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.attribue.name"),
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.Category.Sorting.name"),
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.attribue.desc"),
+                           "getAttribute", "setAttribute", LayoutComboBoxEditor.class));
+         properties.add(LayoutProperty.createProperty(
+                           this, CircularDirection.class,
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.NodePlacement.Direction.name"),
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.Category.NodePlacement.name"),
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.NodePlacement.Direction.desc"),
+                           "getNodePlacementDirection", "setNodePlacementDirection", RotationComboBoxEditor.class));
+         properties.add(LayoutProperty.createProperty(
+                           this, Boolean.class,
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.Transition.EnableTransition.name"),
+                           TRANSITION_CATEGORY,
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.Transition.EnableTransition.desc"),
+                           "isNodePlacementTransition", "setNodePlacementTransition"));
+         properties.add(LayoutProperty.createProperty(
+                           this, Double.class,
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.Transition.TransitionSteps.name"),
+                           TRANSITION_CATEGORY,
+                           NbBundle.getMessage(getClass(), "DualCircleLayout.Transition.TransitionSteps.desc"),
+                           "getTransitionSteps", "setTransitionSteps"));
+      }
+      catch (Exception e) {
+         e.printStackTrace();
+      }
+      return properties.toArray(new LayoutProperty[0]);
+   }
+
+   @Override
+   public void resetPropertiesValues()
+   {
+      setInnerNodeCount(4);
+      setHighDegreeOutside(false);
+      setNodePlacementDirection(CircularDirection.CCW);
+      setAttribute("NodeID");
+      setNodePlacementTransition(false);
+      setTransitionSteps(100000.0);
+   }
+
+   public void setInnerNodeCount(Integer intsecondarynodecount)
+   {
+      GraphController graphController = Lookup.getDefault().lookup(GraphController.class);
```

Comment:
We need to check but calling the `graphController.getModel()` might now work when opening a .gephi project as a current workspace isn't set yet, similar to the issue described here https://github.com/gephi/gephi/issues/1373